cns/restserver/util.go

```diff
@@ -350,11 +359,149 @@ func (service *HTTPRestService) removeToBeDeletedIPStateUntransacted(
+					getNetworkContainersResponse = append(getNetworkContainersResponse, getNetworkContainerResponse)
+					return getNetworkContainersResponse
+				}
+
+				vfpUpdateComplete := !waitingForUpdate
+				ncstatus := service.state.ContainerStatus[ncid]
+				// Update the container status if-
+				// 1. VfpUpdateCompleted successfully
+				// 2. VfpUpdateComplete changed to false
+				if (getNetworkContainerResponse.Response.ReturnCode == types.NetworkContainerVfpProgramComplete &&
+					vfpUpdateComplete && ncstatus.VfpUpdateComplete != vfpUpdateComplete) ||
+					(!vfpUpdateComplete && ncstatus.VfpUpdateComplete != vfpUpdateComplete) {
+					logger.Printf("[Azure-CNS] Setting VfpUpdateComplete to %t for NCID: %s", vfpUpdateComplete, ncid)
+					ncstatus.VfpUpdateComplete = vfpUpdateComplete
+					service.state.ContainerStatus[ncid] = ncstatus
+					if err = service.saveState(); err != nil {
+						logger.Errorf("Failed to save goal states due to %s", err)
+					}
+				}
+			}
+		} else if service.ChannelMode == cns.Managed {
+			// If the NC goal state doesn't exist in CNS running in managed mode, call DNC to retrieve the goal state
+			var (
+				dncEP     = service.GetOption(acn.OptPrivateEndpoint).(string)
+				infraVnet = service.GetOption(acn.OptInfrastructureNetworkID).(string)
+				nodeID    = service.GetOption(acn.OptNodeID).(string)
+			)
+
+			service.Unlock()
+			getNetworkContainerResponse.Response.ReturnCode, getNetworkContainerResponse.Response.Message = service.SyncNodeStatus(dncEP, infraVnet, nodeID, req.OrchestratorContext)
+			service.Lock()
+			if getNetworkContainerResponse.Response.ReturnCode == types.NotFound {
+				getNetworkContainersResponse = append(getNetworkContainersResponse, getNetworkContainerResponse)
+				return getNetworkContainersResponse
+			}
+			networkContainerID = string(ncidBytes)
+		}
+
+		logger.Printf("networkContainerID string %s", networkContainerID)
+
+	default:
+		getNetworkContainerResponse.Response.ReturnCode = types.UnsupportedOrchestratorType
+		getNetworkContainerResponse.Response.Message = fmt.Sprintf("Invalid orchestrator type %v", service.state.OrchestratorType)
+		getNetworkContainersResponse = append(getNetworkContainersResponse, getNetworkContainerResponse)
+		return getNetworkContainersResponse
+	}
+
+	// get ncSet from networkContainerID and unmarshal to list, i.e.,["nc1","nc2"]
+	ncSet := &Set{}
+	if err := json.Unmarshal([]byte(networkContainerID), &ncSet); err != nil {
+		fmt.Println("Failed to unmarshal NC set")
+	}
+
+	for ncid := range *ncSet {
+		containerStatus := service.state.ContainerStatus
+		containerDetails, ok := containerStatus[ncid]
+		if !ok {
+			getNetworkContainerResponse.Response.ReturnCode = types.UnknownContainerID
+			getNetworkContainerResponse.Response.Message = "NetworkContainer doesn't exist."
+			getNetworkContainersResponse = append(getNetworkContainersResponse, getNetworkContainerResponse)
+			return getNetworkContainersResponse
```

Comment:
instead of returning immediately, we should add all the responses and then return.